site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,152 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import FormHelperText from "@mui/material/FormHelperText";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField, { type TextFieldProps } from "@mui/material/TextField";
+import { useState, type FC, useEffect } from "react";
+import {
+  type TimeUnit,
+  durationInDays,
+  durationInHours,
+  suggestedTimeUnit,
+} from "utils/time";
+
+type DurationFieldProps = Omit<TextFieldProps, "value" | "onChange"> & {
+  // Value is in ms
+  value: number;
```

Comment:
For clarity, do you think it'd help to change the prop name to `valueMs`? It's only two more characters, but I feel like it'd remove the need for someone to open the file to see what kind of number format the component expects